packages/eslint-plugin/tests/rules/class-literals-style.test.ts

```diff
@@ -0,0 +1,261 @@
+            return 'a literal value';
+          }
+        }
+      `,
+      output: `
+        class Mx {
+          public readonly [myValue]='a literal value'
+        }
+      `,
+      errors: [
+        {
+          messageId: 'preferFieldStyle',
+        },
+      ],
+      options: ['fields'],
+    },
+    {
+      code: `
+        class Mx {
+          public get [myValue]() {
+            return 12345n;
+          }
+        }
+      `,
+      output: `
+        class Mx {
+          public readonly [myValue]=12345n
+        }
+      `,
+      errors: [
+        {
+          messageId: 'preferFieldStyle',
+        },
+      ],
+      options: ['fields'],
+    },
+    {
+      code: `
+        class Mx {
+          public readonly [myValue] = 'a literal value';
+        }
+      `,
+      output: `
+        class Mx {
+          public get [myValue](){return 'a literal value'}
+        }
+      `,
+      errors: [
+        {
+          messageId: 'preferGetterStyle',
+        },
+      ],
+      options: ['getters'],
+    },
+    {
+      code: 'class Mx { readonly p1 = "hello world"; }',
+      output: 'class Mx { get p1(){return "hello world"} }',
+      errors: [
+        {
+          messageId: 'preferGetterStyle',
+        },
+      ],
+      options: ['getters'],
+    },
+    {
+      code: 'class Mx { readonly p1 = `hello world`; }',
+      output: 'class Mx { get p1(){return `hello world`} }',
+      errors: [
+        {
+          messageId: 'preferGetterStyle',
+        },
+      ],
+      options: ['getters'],
+    },
+    {
+      code: 'class Mx { static readonly p1 = "hello world"; }',
+      output: 'class Mx { static get p1(){return "hello world"} }',
+      errors: [
+        {
+          messageId: 'preferGetterStyle',
+        },
+      ],
+      options: ['getters'],
+    },
+    {
+      code: 'class Mx { protected get p1() { return "hello world"; } }',
+      output: 'class Mx { protected readonly p1="hello world" }',
+      errors: [
+        {
+          messageId: 'preferFieldStyle',
+        },
+      ],
+      options: ['fields'],
+    },
+    {
+      code: 'class Mx { protected readonly p1 = "hello world"; }',
+      output: 'class Mx { protected get p1(){return "hello world"} }',
+      errors: [
+        {
+          messageId: 'preferGetterStyle',
+        },
+      ],
+      options: ['getters'],
+    },
+    {
+      code: 'class Mx { public static get p1() { return "hello world"; } }',
+      output: 'class Mx { public static readonly p1="hello world" }',
+      errors: [
+        {
+          messageId: 'preferFieldStyle',
+        },
+      ],
+      options: ['fields'],
+    },
+    {
+      code: 'class Mx { public static readonly p1 = "hello world"; }',
+      output: 'class Mx { public static get p1(){return "hello world"} }',
+      errors: [
+        {
+          messageId: 'preferGetterStyle',
+        },
+      ],
+      options: ['getters'],
+    },
+  ],
```

Comment:
it's completely stupid and absurd, but the following is syntactically valid:

```ts
class Foo {
  public static readonly static private public protected get foo() { return 1; }
}
```

for sanity, could you add an error test for this to ensure that the fixer outputs syntactically and semantically valid code in this case?